mlflow/store/ftp_artifact_repo.py

```diff
@@ -50,7 +50,9 @@ def _is_dir(self, full_file_path):
     def _mkdir(self, artifact_dir):
         with self.get_ftp_client() as ftp:
             try:
-                ftp.mkd(artifact_dir)
+                head, _ = posixpath.split(artifact_dir)
+                if artifact_dir not in ftp.nlst(head):
```

Comment:
I couldn't find a super obvious API, but is there no standard way to stat for existence of a specific file, as opposed to listing all files in the directory?

Reply:
I also wondered about that but as it seems ftplib (or ftp itself) doesn't provide a way to check for existence of a folder. Apparently using `nlst` is the only way to realize this.

Reply:
I actually just realized that there is a `_is_dir` function, shame on me. Now using that one :smile: